File: src/TradingSystem.Web/Services/MarketApiService.cs
using System.Net.Http.Json;
using TradingSystem.Web.Models;

namespace TradingSystem.Web.Services;

public interface IMarketApiService
{
    Task<StockQuote?> GetQuoteAsync(string symbol);
    Task<TechnicalIndicators?> GetIndicatorsAsync(string symbol);
    Task<List<StockData>?> GetHistoryAsync(string symbol, int days = 30);
}

public class MarketApiService : IMarketApiService
{
    private readonly HttpClient _http;

    public MarketApiService(HttpClient http)
    {
        _http = http;
    }

    public async Task<StockQuote?> GetQuoteAsync(string symbol)
    {
        var response = await _http.GetFromJsonAsync<ApiResponse<StockQuote>>($"market/{symbol}");
        return response?.Data;
    }

    public async Task<TechnicalIndicators?> GetIndicatorsAsync(string symbol)
    {
        var response = await _http.GetFromJsonAsync<ApiResponse<TechnicalIndicators>>($"market/{symbol}/indicators");
        return response?.Data;
    }

    public async Task<List<StockData>?> GetHistoryAsync(string symbol, int days = 30)
    {
        var response = await _http.GetFromJsonAsync<ApiResponse<List<StockData>>>($"market/{symbol}/history?days={days}");
        return response?.Data;
    }
}

File: src/TradingSystem.Angular/src/app/services/market-api.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable, of } from 'rxjs';
import { map, catchError } from 'rxjs/operators';
import { ApiResponse, StockQuote, TechnicalIndicators, StockData } from '../models/models';

@Injectable({ providedIn: 'root' })
export class MarketApiService {
  constructor(private http: HttpClient) {}

  getQuote(symbol: string): Observable<StockQuote | null> {
    return this.http.get<ApiResponse<StockQuote>>(`/market/${symbol}`).pipe(
      map(r => r.data || null),
      catchError(() => of(null))
    );
  }

  getIndicators(symbol: string): Observable<TechnicalIndicators | null> {
    return this.http.get<ApiResponse<TechnicalIndicators>>(`/market/${symbol}/indicators`).pipe(
      map(r => r.data || null),
      catchError(() => of(null))
    );
  }

  getHistory(symbol: string, days: number = 30): Observable<StockData[]> {
    return this.http.get<ApiResponse<StockData[]>>(`/market/${symbol}/history?days=${days}`).pipe(
      map(r => r.data || []),
      catchError(() => of([]))
    );
  }
}
